SDL/NeuralNetwork.cuh

```diff
@@ -0,0 +1,175 @@
+
+        // Build DNN input vector (corresponding output N-tuple branch noted in parenthetical in comment)
+        float x[38] = {
+            log10(2*SDL::k2Rinv1GeVf*innerRadius),                                       // inner T3 pT (t3_pt)
+            mdsInGPU.anchorEta[mdIndex1],                                                // inner T3 anchor hit 1 eta (t3_0_eta)
+            mdsInGPU.anchorPhi[mdIndex1],                                                // inner T3 anchor hit 1 phi (t3_0_phi)
+            mdsInGPU.anchorZ[mdIndex1],                                                  // inner T3 anchor hit 1 z (t3_0_z)
+            sqrtf(x1*x1 + y1*y1),                                                        // inner T3 anchor hit 1 r (t3_0_r)
+            float(modulesInGPU.layers[lowerModuleIndex1] + 6*is_endcap1),                // inner T3 anchor hit 1 layer (t3_0_layer)
+            mdsInGPU.anchorEta[mdIndex2],                                                // inner T3 anchor hit 2 eta (t3_2_eta)
+            mdsInGPU.anchorPhi[mdIndex2],                                                // inner T3 anchor hit 2 phi (t3_2_phi)
+            mdsInGPU.anchorZ[mdIndex2],                                                  // inner T3 anchor hit 2 z (t3_2_z)
+            sqrtf(x2*x2 + y2*y2),                                                        // inner T3 anchor hit 2 r (t3_2_r)
+            float(modulesInGPU.layers[lowerModuleIndex2] + 6*is_endcap2),                // inner T3 anchor hit 2 layer (t3_2_layer)
+            mdsInGPU.anchorEta[mdIndex3],                                                // inner T3 anchor hit 3 eta (t3_4_eta)
+            mdsInGPU.anchorPhi[mdIndex3],                                                // inner T3 anchor hit 3 phi (t3_4_phi)
+            mdsInGPU.anchorZ[mdIndex3],                                                  // inner T3 anchor hit 3 z (t3_4_z)
+            sqrtf(x3*x3 + y3*y3),                                                        // inner T3 anchor hit 3 r (t3_4_r)
+            float(modulesInGPU.layers[lowerModuleIndex3] + 6*is_endcap3),                // inner T3 anchor hit 3 layer (t3_4_layer)
+            log10(2*SDL::k2Rinv1GeVf*outerRadius),                                       // outer T3 pT (t3_pt)
+            mdsInGPU.anchorEta[mdIndex3],                                                // outer T3 anchor hit 4 eta (t3_0_eta)
+            mdsInGPU.anchorPhi[mdIndex3],                                                // outer T3 anchor hit 4 phi (t3_0_phi)
+            mdsInGPU.anchorZ[mdIndex3],                                                  // outer T3 anchor hit 3 eta (t3_0_z)
+            sqrtf(x3*x3 + y3*y3),                                                        // outer T3 anchor hit 3 r (t3_0_r)
+            float(modulesInGPU.layers[lowerModuleIndex3] + 6*is_endcap3),                // outer T3 anchor hit 3 layer (t3_0_layer)
```

Comment:
@jkguiang 
is this intentionally identical to L85-90?
we could've saved a bit on the number of weights, matrix operations, and having the network to learn identities.

Please check and perhaps keep a todo somewhere to possibly clean this up

Reply:
It was not a design decision, but it is something we could indeed clean up.